pre-commit/build_docs.py

```diff
@@ -0,0 +1,147 @@
+"""
+Builds connector detail documentation similar to
+https://my.phantom.us/<soar_version>/docs/app_reference/<connector>,
+in github-flavored markdown format, and combines it with human-written
+author notes (legacy readme.md).
+"""
+import argparse
+import json
+import logging
+import os
+import re
+import sys
+from pathlib import Path
+
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+from jinja2.ext import Extension
+from jinja2.lexer import Token
+
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+TEMPLATE_DIR = Path(SCRIPT_DIR, "templates")
+TEMPLATE_NAME = "connector_detail.md"
+README_OUTPUT_NAME = "readme.md"
+README_INPUT_NAME = "author_notes.md"
+
+# From https://enterprise.github.com/downloads/en/markdown-cheatsheet.pdf
+MARKDOWN_RESERVED_CHARACTERS = ["\\", "`", "*", "_", "{", "}", "[", "]", "(",
+                                ")", "#", "+", "-", ".", "!",  "@", "|", ":"]
+
+
+class EscapeMDFilterVarsExtension(Extension):
+    # Markdown in all unfiltered injected values should be escaped
+    def filter_stream(self, stream):
+        prev_token = None
+        for token in stream:
+            if (token.type == "variable_end"
+                    and str(prev_token) not in self.environment.filters):
+                yield Token(token.lineno, "pipe", "|")
+                yield Token(token.lineno, "name", "escape_markdown")
+            yield token
+            prev_token = token
+
+
+def generate_gh_fragment(text):
+    text = text.lower()
+    text = re.sub(r"[^a-zA-Z0-9\s-]", "", text)
+    return re.sub(r"[\s]", "-", text)
+
+
+def generate_action_heading_text(text):
+    return f"action: '{text}'"
+
+
+def escape_markdown(text):
+    if isinstance(text, str):
+        for reserved_char in MARKDOWN_RESERVED_CHARACTERS:
+            text = text.replace(reserved_char, f"\\{reserved_char}")
+    return text
+
+
+def get_app_json(app_json_dir):
+    logging.info("Looking for app JSON in: %s", app_json_dir)
+    app_json_name = [f for f in os.listdir(app_json_dir)
+                     if f.endswith(".json")
+                     and "postman_collection" not in f][0]
+    json_file_path = Path(app_json_dir, app_json_name)
+    logging.info("Loading json: %s", app_json_name)
+    with open(json_file_path, "r") as json_file:
+        return json.load(json_file)
+
+
+def load_or_rename_input_readme(connector_path):
```

Comment:
this function name implies the input readme (`author_notes.md`) is being re-named but we're actually renaming a legacy version of the output readme (`readme.md`) to the input readme

would it be a bit more clear to break this function up?